### PR TITLE
Refactor trakt plugins

### PR DESCRIPTION
**Pull request by cvium**
### Motivation for changes:
Just wanted to see if it could be made more readable
### Detailed changes:
- Extracted a lot of conditional statements into separate functions

#### To Do:

- [x] Stop long running transactions
- [x] trakt_list
- [x] next_trakt_episodes

### CODE REVIEW

**Comment by cvium** [comment]
This is just an attempt at making the code less confusing to read. Not entirely sure I succeeded. It's a trade-off with no clear winner.

File: flexget/plugins/metainfo/trakt_lookup.py [hunk 1]
```diff
@@ -181,214 +180,250 @@ class PluginTraktLookup(object):
...
+    def lazy_movie_user_ratings_lookup(self, entry):
+        with Session() as session:
+            try:
+                entry['trakt_movie_user_rating'] = self.trakt.movie_user_ratings(self.__get_movie(entry, session),
+                                                                                 entry['title'])
+            except LookupError as e:
+                log.debug(e)
+        return entry
+
+    # TODO: these shouldn't be here?
```
Comment (liiight) [inline flexget/plugins/metainfo/trakt_lookup.py]: yeah, could move it to be under `entry`

File: flexget/plugins/metainfo/trakt_lookup.py [hunk 2]
```diff
@@ -181,214 +180,250 @@ class PluginTraktLookup(object):
...
-                                                         account=config.get('account'))
-                    entry['trakt_movie_user_rating'] = movie_rating
+                entry['trakt_collected'] = self.trakt.is_episode_in_collection(self.__get_episode(entry, session),
+                                                                               entry['title'])
             except LookupError as e:
                 log.debug(e)
         return entry
 
+    def lazy_movie_collected_lookup(self, entry):
+        with Session() as session:
```
Comment (liiight) [inline flexget/plugins/metainfo/trakt_lookup.py]: all of these lookup method are almost identical in structure, maybe create a generic one and use that?

File: flexget/plugins/metainfo/trakt_lookup.py [hunk 3]
```diff
@@ -399,41 +434,53 @@ def on_task_metainfo(self, task, config):
...
+                self.__register_lazy_user_ratings_lookup(entry)
+
+    def __register_lazy_collected_lookup(self, entry):
+        if self.__entry_is_episode(entry):
+            entry.register_lazy_func(self.lazy_episode_collected_lookup, ['trakt_collected'])
+        elif self.__entry_is_season(entry):
+            entry.register_lazy_func(self.lazy_season_collected_lookup, ['trakt_collected'])
+        elif self.__entry_is_show(entry):
+            entry.register_lazy_func(self.lazy_show_collected_lookup, ['trakt_collected'])
+        else:
```
Comment (liiight) [inline flexget/plugins/metainfo/trakt_lookup.py]: maybe it's worth adding an implicit check for movie type at this opportunity?

File: flexget/plugins/internal/api_trakt.py [hunk 4]
```diff
@@ -843,236 +842,358 @@ def __init__(self, search, movie_id=None, movie=None):
...
-                cache[movie_id]['watched_at'] = dateutil_parse(movie['last_watched_at'], ignoretz=True)
-                cache[movie_id]['plays'] = movie['plays']
-        else:
-            for series in data:
-                series_id = series['show']['ids']['trakt']
-                cache[series_id] = series['show']
-                cache[series_id]['seasons'] = series['seasons']
-                cache[series_id]['watched_at'] = dateutil_parse(series['last_watched_at'], ignoretz=True)
-                cache[series_id]['plays'] = series['plays']
+        return get_session().get(get_api_url(media_type + 's', trakt_id), params={'extended': 'full'}).json()
```
[the remark this answers is not in the document]
Comment (cvium) [reply flexget/plugins/internal/api_trakt.py]: It's only one parameter though

File: flexget/entry.py [hunk 5]
```diff
@@ -184,6 +184,22 @@ def failed(self):
     def undecided(self):
         return self._state == 'undecided'
 
+    @property
```
Comment (paranoidi) [inline flexget/entry.py]: entry is supposed to be content agnostic

**Comment by mfonville** [comment]
Please check the bug I hit in https://github.com/Flexget/Flexget/issues/2161 and probable fix

**Pull request merged**